pkg/util/languagedetection/languageset.go

```diff
@@ -0,0 +1,62 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package languagedetection
+
+import (
+	"sort"
+	"strings"
+)
+
+// LanguageSetInterface is an interface defining the behaviour of a language set
```

Comment:
Same here, it's way overkill to define all that for what's basically a `set<string>`. You could `type LanguageSet map[string]struct{}` if you really want a type but not sure it's necessary for something that simple.